src/core/core.scale.js

```diff
@@ -630,9 +630,14 @@ module.exports = Element.extend({
 		var isHorizontal = me.isHorizontal();
 		var optionTicks = me.options.ticks.minor;
 		var tickCount = ticks.length;
+
+		// take the largest of either the width or height when rotated
+		// when at 90deg then the height of the text is its width
 		var labelRotationRadians = helpers.toRadians(me.labelRotation);
-		var cosRotation = Math.cos(labelRotationRadians);
-		var longestRotatedLabel = me.longestLabelWidth * cosRotation;
+		var cosRotation = Math.abs(Math.cos(labelRotationRadians));
+		var sinRotation = Math.abs(Math.sin(labelRotationRadians));
+		var longestRotatedLabel = Math.max(me.longestLabelWidth * cosRotation, me.options.ticks.fontSize * sinRotation);
```

Comment:
I'm not sure that's correct (neither the original formula) because it works only with a line (width *or* height), not a box (width *and* height). When angle is not 0, 90, 180, 270, we should take in account both dimensions:

![image](https://user-images.githubusercontent.com/3874900/43687906-a137f8cc-98de-11e8-98c5-666ef0dd58e8.png)

```javascript
var rot = helpers.toRadians(me.labelRotation);
var cos = Math.abs(Math.cos(rot));
var sin = Math.abs(Math.sin(rot));
var width = me.longestLabelWidth; 
var height = me.options.ticks.fontSize;
var skipSize =  (width * cos) + (height * sin)

// In case we want to support auto-skip on vertical scales
// var skipSize =  (width * sin) + (height * cos)
```

Also, `me.options.ticks.fontSize` works only for single line label, should we also support multilines?

Reply:
My suggested `skipSize` wasn't optimal, correct formula in [this pen](https://codepen.io/simonbrunel/pen/VqjBmx), implemented in #5922.